bin/BuildPackages.sh

```diff
@@ -79,115 +135,197 @@ build_carat() {
     else
-      ./configure $GAPDIR
-    fi;
-    $MAKE
+      ./configure "$GAPDIR"
+    fi
+    "$MAKE"
   else
-    echo "No building required for $dir"
-  fi;
+    notice "No building required for $PKG"
+  fi
 }
 
-for dir in `ls -d */`
+build_one_package() {
+  # requires one argument which is the package directory
+  PKG="$1"
+  echo ""
+  date
+  echo ""
+  notice "==== Checking $PKG"
+  (  # start subshell
+  set -e
+  cd "$CURDIR/$PKG"
+  case "$PKG" in
+    anupq*)
+      ./configure "CFLAGS=-m32 LDFLAGS=-m32 --with-gaproot=$GAPDIR" && \
+      "$MAKE" CFLAGS=-m32 LOPTS=-m32
+    ;;
+
+    atlasrep*)
+      chmod 1777 datagens dataword
+    ;;
+
+    carat*)
+      build_carat
+    ;;
+
+    cohomolo*)
+      build_cohomolo
+    ;;
+
+    fplsa*)
+      ./configure "$GAPDIR" && \
+      "$MAKE" CC="gcc -O2 "
+    ;;
+
+    kbmag*)
+      ./configure "$GAPDIR" && \
+      "$MAKE" COPTS="-O2 -g"
+    ;;
+
+    NormalizInterface*)
+      ./build-normaliz.sh "$GAPDIR" && \
+      run_configure_and_make
+    ;;
+
+    pargap*)
+      ./configure "$GAPDIR" && \
+      "$MAKE" && \
+      cp bin/pargap.sh "$GAPDIR/bin" && \
+      rm -f ALLPKG
+    ;;
+
+    xgap*)
+      ./configure --with-gaproot="$GAPDIR" && \
+      "$MAKE" && \
+      rm -f "$GAPDIR/bin/xgap.sh" && \
+      cp bin/xgap.sh "$GAPDIR/bin"
+    ;;
+
+    simpcomp*)
+    ;;
+
+    *)
+      run_configure_and_make
+    ;;
+  esac
+  ) || build_fail
+}
+
+date >> "$LOGDIR/$FAILPKGFILE.log"
+for PKG in "${PACKAGES[@]}"
 do
-    if [ -e $dir/PackageInfo.g ]; then
-      dir="${dir%/}"
-      echo "==== Checking $dir"
-      (  # start subshell
-      set -e
-      cd $dir
-      case $dir in
-        anupq*)
-          ./configure CFLAGS=-m32 LDFLAGS=-m32 --with-gaproot=$GAPDIR &&
-          $MAKE CFLAGS=-m32 LOPTS=-m32
-        ;;
-
-        atlasrep*)
-          chmod 1777 datagens dataword
-        ;;
-
-        carat*)
-          build_carat
-        ;;
-
-        cohomolo*)
-          build_cohomolo
-        ;;
-
-        fplsa*)
-          ./configure $GAPDIR &&
-          $MAKE CC="gcc -O2 "
-        ;;
-
-        kbmag*)
-          ./configure $GAPDIR &&
-          $MAKE COPTS="-O2 -g"
-        ;;
-
-        NormalizInterface*)
-          ./build-normaliz.sh $GAPDIR &&
-          run_configure_and_make
-        ;;
-
-        pargap*)
-          ./configure $GAPDIR &&
-          $MAKE &&
-          cp bin/pargap.sh $GAPDIR/bin &&
-          rm -f ALLPKG
-        ;;
-
-        xgap*)
-          ./configure --with-gaproot=$GAPDIR &&
-          $MAKE &&
-          rm -f $GAPDIR/bin/xgap.sh &&
-          cp bin/xgap.sh $GAPDIR/bin
-        ;;
-
-        simpcomp*)
-        ;;
-        
-        *)
-          run_configure_and_make
-        ;;
-      esac;
-      ) || build_fail
-      # end subshell
-    else
-      echo "$dir is not a GAP package -- no PackageInfo.g"
-    fi;
-done;
+  PKG="${PKG%/}"
+  PKG="${PKG##*/}"
```

Comment:
What do these two lines do? Add a comment (not everybody is a bash expert, and this script needs to b maintainable by as many people as possibl).